nntrainer/src/tensor.cpp

```diff
@@ -592,143 +592,118 @@ Tensor Tensor::sum(int axis) const {
   return ret;
 }
 
-/**
- * If the dim.batch() size of m is one, the it is reused for
- * every calculation along with dim.batch().
- * TODO: support dim.channel() > 1.
- */
-Tensor Tensor::dot(Tensor const &m) const {
-  if (dim.width() != m.dim.height()) {
-    throw std::runtime_error("Error dim.width() != m.dim.height()");
+Tensor Tensor::dot(Tensor const &m, bool trans, bool trans_m) const {
+  if (m.dim.rank() > 2) {
```

Comment:
We discussed offline to add safeguard to `when (this->dim.rank() > 2)` in case of `trans` is true

Reply:
done.